internal/session/handle.go

```diff
@@ -221,11 +221,19 @@ func (s *Session) handleSelectedCommand(
 
 	return s.state.Selected(ctx, func(mailbox *state.Mailbox) error {
 		okResponse, err := s.handleWithMailbox(ctx, tag, cmd, mailbox, ch)
-		if err != nil {
-			return err
+
+		// Allow state updates to be applied if the command failed. It might resolve
+		// some invalid state problem.
+		flushErr := flush(ctx, mailbox, false, ch)
+		if flushErr != nil {
+			if err != nil {
+				return fmt.Errorf("%w (flush err:%v)", err, flushErr)
+			}
+
+			return flushErr
 		}
```

Comment:
```suggestion
		// Allow state updates to be applied if the command failed. It might resolve
		// some invalid state problem.
		if flushErr := flush(ctx, mailbox, false, ch); flushErr != nil {
			if err != nil {
				return fmt.Errorf("%w (flush err: %v)", err, flushErr)
			}

			return flushErr
		}
```

since we have multiple errors here, it would be nice to scope the flush error just to this block so it doesn't get confusing

Reply:
Done